### PR TITLE
Generate consistency proofs in test vectors

### PR DESCRIPTION
**Pull request by jtoohill**
The trusted log root will fall behind a bit to make consistency proofs
more interesting.

Also changes the testdata generator to only open and write files once,
rather than overwriting the same file multiple times.

### CODE REVIEW

File: cmd/gen-test-vectors/main.go [hunk 1]
```diff
@@ -143,16 +146,19 @@ func GenerateTestVectors(ctx context.Context, env *integration.Env) error {
...
+			UserId:        tc.userID,
+			AppId:         appID,
+			FirstTreeSize: int64(slr.TreeSize),
 		})
 		if err != nil {
 			return fmt.Errorf("gen-test-vectors: GetEntry(): %v", err)
 		}
-		if _, _, err := env.Client.VerifyGetEntryResponse(ctx, env.Domain.DomainId, appID, tc.userID, types.LogRootV1{}, e); err != nil {
+		var newslr *types.LogRootV1
+		if _, newslr, err = env.Client.VerifyGetEntryResponse(ctx, env.Domain.DomainId, appID, tc.userID, *slr, e); err != nil {
```
Comment (golangcibot) [inline cmd/gen-test-vectors/main.go]: line is 122 characters

File: cmd/gen-test-vectors/main.go [hunk 2]
```diff
@@ -163,6 +169,12 @@ func GenerateTestVectors(ctx context.Context, env *integration.Env) error {
 			Resp:   e,
 		})
 
+		// Let the trusted log root fall behind by a few revisions
+		// to make consistency proofs more interesting.
+		if newslr.TreeSize%5 == 1 {
```
Comment (gdbelvin) [inline cmd/gen-test-vectors/main.go]: nit: might be cleaner to compare with 0 -- unless you're using 1 for a reason in which case it would be nice to document it.

Comment (jtoohill) [reply cmd/gen-test-vectors/main.go]: Documented that we're using 1 intentionally. (There aren't more than 5 test cases at the moment, so 0 would use the empty trusted log root which doesn't require consistency proofs.)

File: core/client/verify_test.go [hunk 3]
```diff
@@ -58,12 +58,16 @@ func TestVerifyGetEntryResponse(t *testing.T) {
 		t.Fatalf("Unmarshal(): %v", err)
 	}
 
+	trusted := &types.LogRootV1{}
 	for _, tc := range getEntryResponses {
 		t.Run(tc.Desc, func(t *testing.T) {
-			trusted := types.LogRootV1{}
-			if _, _, err := v.VerifyGetEntryResponse(ctx, domainPB.DomainId, tc.AppID, tc.UserID, trusted, tc.Resp); err != nil {
+			var slr *types.LogRootV1
+			if _, slr, err = v.VerifyGetEntryResponse(ctx, domainPB.DomainId, tc.AppID, tc.UserID, *trusted, tc.Resp); err != nil {
```
Comment (golangcibot) [inline core/client/verify_test.go]: line is 122 characters
